kart/data.py

```diff
@@ -28,27 +28,44 @@ def data(ctx, **kwargs):
     type=click.Choice(["text", "json"]),
     default="text",
 )
+@click.option(
+    "--with-dataset-types/--without-dataset-types",
+    is_flag=True,
+    help="When set, outputs the dataset type and version. (This may become the default in a later version of Kart)",
+)
 @click.argument("refish", required=False, default="HEAD", shell_complete=ref_completer)
 @click.pass_context
-def data_ls(ctx, output_format, refish):
+def data_ls(ctx, output_format, with_dataset_types, refish):
     """List all of the datasets in the Kart repository"""
     repo = ctx.obj.get_repo(allowed_states=KartRepoState.ALL_STATES)
-    ds_paths = list(repo.datasets(refish).paths())
+    json_list = [
+        {"path": ds.path, "type": ds.DATASET_TYPE, "version": ds.VERSION}
+        for ds in repo.datasets(refish)
+    ]
+
+    if output_format == "text" and not json_list:
+        repo_desc = (
+            "Empty repository."
+            if repo.head_is_unborn
+            else "The commit at HEAD has no datasets."
+        )
+        click.echo(f'{repo_desc}\n  (use "kart import" to add some data)')
+        return
 
     if output_format == "text":
-        if ds_paths:
-            for ds_path in ds_paths:
-                click.echo(ds_path)
+        if with_dataset_types:
+            for ds_obj in json_list:
+                click.echo(f"{ds_obj['path']}\t({ds_obj['type']}.v{ds_obj['version']})")
```

Comment:
strings inside strings are hard to read, pull this out somehow. perhaps don't use an f-string, use a `.format(**ds_obj)`